lib/mongoid/association/accessors.rb

```diff
@@ -43,7 +43,17 @@ def __build__(name, object, association, selected_fields = nil)
       def create_relation(object, association, selected_fields = nil)
         type = @attributes[association.inverse_type]
         target = association.build(self, object, type, selected_fields)
-        target ? association.create_relation(self, target) : nil
+        target = target ? association.create_relation(self, target) : nil
+
+        # Only need to do this on embedded associations. The pending callbacks
+        # are only added when materializing the documents, which only happens
+        # on embedded associations. There is no call to the database in the
+        # construction of a referenced association.
```

Comment:
This comment goes with the condition on line 54, right? If so can this `if` statement also be turned around so that the condition is on top next to the comment.

Reply:
done